client/src/components/views/Orders.jsx

```diff
@@ -0,0 +1,236 @@
+  const [deliveryDate, setDeliveryDate] = useState();
+  const [orderDate, setOrderDate] = useState();
+  const [quantity, setQuantity] = useState(0);
+  const manufacturer = useRef('');
+  const location = useRef('');
+  const note = useRef('');
+  const materialTypes = ['rubber', 'aluminum', 'steel', 'copper', 'plastic', 'leather'];
+  const materialCost = useQuery('orders/materialList', getMaterialList);
+
+  const handleChangePage = (event, newPage) => {
+    setPage(newPage);
+  };
+
+  const handleChangeRowsPerPage = (event) => {
+    setRowsPerPage(parseInt(event.target.value, 10));
+    setPage(0);
+  };
+
+  // sets material and cost states
+  const handleMaterial = (e) => {
+    const choice = e.target.value;
+    setMaterial(choice);
+    setCost(materialCost.data.data[choice].cost);
+  };
+
+  // generates a random int within ranges
+  function getRandomInt(min, max) {
+    return Math.floor(Math.random() * (max - min)) + min;
+  }
+
+  const handleSubmit = async () => {
+    const orderTime = new Date();
+    const deliveryTime = new Date();
+    const randomDeliveryDate = deliveryTime.getDate() + getRandomInt(1, 8); // random delivery date within a week
+    deliveryTime.setDate(randomDeliveryDate);
+
+    setOrderDate(orderTime);
+    setDeliveryDate(deliveryTime);
+
+    try {
+      await postOrders({
+        materialType: material,
+        cost: cost,
+        quantity: quantity,
+        deliveryDate: deliveryDate,
+        orderDate: orderDate,
+        manufacturerName: manufacturer.current.value,
+        vendorLocation: location.current.value,
+        status: 'Pending',
+        note: note.current.value,
+      });
+      toast({
+        title: 'Order Placed',
+        description: 'The order has been placed',
+        status: 'success',
+        duration: 9000,
+        isClosable: true,
+      });
+    } catch {
+      toast({
+        position: 'top',
+        title: 'An error occurred.',
+        description: 'Unable to place order.',
+        status: 'error',
+        duration: 2000,
+        isClosable: true,
+      });
+    }
+    refetch();
+    onClose();
+  };
+
+  const openOrderForm = () => (
+    <Modal isOpen={isOpen} onClose={onClose}>
+      <ModalOverlay />
+      <ModalContent>
+        <ModalHeader>Place an order</ModalHeader>
+        <ModalCloseButton />
+        <ModalBody pb={6}>
+          <FormControl isRequired>
+            <FormLabel>Material</FormLabel>
+            <Select placeholder="Select an option" onChange={handleMaterial}>
+              {materialTypes.map((material) => (
+                <Fragment key={material._id}>
+                  <option value={material}>{material}</option>
+                </Fragment>
+              ))}
+            </Select>
+          </FormControl>
+          <FormControl isRequired>
+            <FormLabel>Quantity</FormLabel>
+            <Input placeholder="Quantity" onChange={(e) => setQuantity(e.target.value)} />
```

Comment:
`Quantity` field should only accept numbers.

Reply:
true! ill change the input field